src/d3a/d3a_core/simulation.py

```diff
@@ -305,12 +305,16 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
             log.debug(f"Used {mbs_used} MBs.")
 
             for tick_no in range(tick_resume, config.ticks_per_slot):
+                tick_start = time.monotonic()
+                while self.paused:
```

Comment:
You need to add some kind of sleep in the while loop, because the current implementation would run the CPU in 100% needlessly. Alternatively, look at the existing implementation which does not exhibit this behavior.